### PR TITLE
Update Dashboard queries

### PR DESCRIPTION
**Pull request by Elkrival**
This pr updates and optimizes the dashboard query by using streams and reducing server load from writing to memory.

### CODE REVIEW

File: server/controllers/dashboardController/index.js [hunk 1]
```diff
@@ -2,36 +2,54 @@ import { ObjectId } from 'mongodb'
...
+      await new Promise((resolve, reject) => {
+        dataStream.on('end', () => resolve())
+
+        dataStream.on('error', (err) => reject(err))
+      })
+
+      const dashboardProcessor = new DashboardDataProcessor(
+        flatConfig,
+        participantDataMap
+      )
```
Comment (mikestone14) [inline server/controllers/dashboardController/index.js]: can we move this logic out of the controller and into the service?

Comment (mikestone14) [reply server/controllers/dashboardController/index.js]: maybe expose a method on the service for `participantDataMap`? Then we can `await service.participantDataMap()`?

Comment (Elkrival) [reply server/controllers/dashboardController/index.js]: [62415d4](https://github.com/AMP-SCZ/dpdash/pull/630/commits/62415d47393169a2b7d7e879b26a8ca7ade93d83)

Comment (Elkrival) [reply server/controllers/dashboardController/index.js]: That suggestion makes sense thanks for that

File: server/controllers/api/assessmentDayDataController/index.js [hunk 2]
```diff
@@ -85,11 +84,9 @@ function sortedDayData(participantAssessmentData, participant_assessments) {
     ({ day }) =>
```
Comment (JackVCurtis) [inline server/controllers/api/assessmentDayDataController/index.js:82]: Is it possible/realistic to do this sorting and filtering in the database query?

Comment (JackVCurtis) [reply server/controllers/api/assessmentDayDataController/index.js:82]: Nevermind, we can optimize on another pass

**Review by JackVCurtis — APPROVED** [review]
💯

**Pull request merged**